### ISSUE TITLE
Public GitRepository using GitLab Requiring Credentials

### ISSUE THREAD

**Issue by ground7** [fluxcd/flux2#2596]
### Describe the bug

Previously I was able to use a gitlab public repository like so:
```yaml
---
apiVersion: source.toolkit.fluxcd.io/v1beta2
kind: GitRepository
metadata:
  name: gitlab-repo
  namespace: flux-system
spec:
  interval: 1h
  url: https://gitlab.com/gitlab-org/gitlab-runner.git
```
This no longer works, I now get:
```
$ flux reconcile source git gitlab-repo
► annotating GitRepository gitlab-repo in flux-system namespace
✔ GitRepository annotated
◎ waiting for GitRepository reconciliation
✗ GitRepository reconciliation failed: 'failed to checkout and determine revision: unable to clone 'https://gitlab.com/gitlab-org/gitlab-runner.git': authentication required'
```
This still works fine for my github sources.

### Steps to reproduce

1. Install flux
2. Add a public GitRepository source with a public gitlab url
3. Try to reconcile

### Expected behavior

Public gitlab repos would work as sources without needing credentials

### OS / Distro

Ubuntu 20.04

### Flux version

v0.28.4

### Flux check

► checking prerequisites
✔ Kubernetes 1.22.7+rke2r2 >=1.20.6-0
► checking controllers
✔ helm-controller: deployment ready
► ghcr.io/fluxcd/helm-controller:v0.18.2
✔ kustomize-controller: deployment ready
► ghcr.io/fluxcd/kustomize-controller:v0.22.2
✔ notification-controller: deployment ready
► ghcr.io/fluxcd/notification-controller:v0.23.1
✔ source-controller: deployment ready
► ghcr.io/fluxcd/source-controller:v0.22.4
✔ all checks passed

### Git provider

GitLab

### Container Registry provider

N/A

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

**Comment by mjnagel** [comment]
I'm seeing this same issue, also with public gitlab repos that were working fine pre-0.28.

**Comment by pjbgf** [comment]
Thank you for reporting this, I have just reproduced this with the default `gitImplementation` which is `go-git`.

Whilst we investigate and resolve this issue, you should be able to make it work by changing the `gitImplementation` to `libgit2`.

**Comment by pjbgf** [comment]
A fix is in the pipeline, we should release a new version tomorrow with it.

**Comment by hiddeco** [comment]
This has been solved in `v0.28.5`, which was released about a minute ago https://github.com/fluxcd/flux2/releases/tag/v0.28.5

**Comment by texasbobs** [comment]
We are experiencing this same issue on v0.28.5 with GitHub.
```
{"level":"error","ts":"2022-04-14T20:51:37.566Z","logger":"controller.gitrepository","msg":"Reconciler error","reconciler group":"source.toolkit.fluxcd.io","reconciler kind":"GitRepository","name":"edge","namespace":"flux-system","error":"failed to checkout and determine revision: unable to clone 'https://<repo-name>': authentication required"}
```

Our secret has the user name and its access token.
